controller/src/main/java/io/pravega/controller/eventProcessor/impl/EventProcessorGroupImpl.java

```diff
@@ -369,6 +364,7 @@ private void replaceCell(String readerId) {
             log.info("Stopping event processor cell: {}", cell);
             try {
                 cell.stopAsync();
+                checkpointStore.removeReader(cell.getProcess(), readerGroup.getGroupName(), readerId);
```

Comment:
why not remove the reader from inside cell.stop which terminates the reader and only after that we remove it. here we could be removing it prematurely.

Reply:
it would make a difference if we are unable to do a clean shutdown (where we use the last reader position and report it offline at that position.. 
say all of that failed but we removed the reader postion from zookeeper, then another controller will not be able to correctly failover this reader.

Reply:
While I agree it would be cleaner, my motivation for not doing it inside the cell, is because it causes problems on shutdown. 
Specifically if the checkpointStore needs to be written to, then every cell will need to timeout writing to it. As there are many cells this results in very long delays in the tests. In fact it is the largest single contributor of integration test time.

I have addressed the concern raised here about this being too soon by moving it after the await.